pixeldata/src/transform.rs

```diff
@@ -189,6 +191,53 @@ fn window_level_sigmoid(value: f64, window_width: f64, window_center: f64, y_max
     y_max / (1. + f64::exp(-4. * (value - wc) / ww))
 }
 
+/// A full description of a VOI LUT function transformation
+/// based on an explicit LUT.
+pub struct VoiLutTransform<'a> {
+    lut: &'a VoiLut,
+    shift: u32,
+}
+
+impl<'a> VoiLutTransform<'a> {
+    /// Create a new LUT transformation.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `lut.data` is empty, or if `bits_stored` is smaller than
+    /// `lut.bits_stored`.
+    pub fn new(lut: &'a VoiLut, bits_stored: u16) -> Self {
+        if lut.data.is_empty() {
+            // we'll do unchecked accesses to the first and last items in apply()
+            panic!("LUT data is empty");
+        }
+
+        if bits_stored < (lut.bits_stored as u16) {
+            panic!(
+                "LUT with BitsStored {} cannot be used for an image with BitsStored {}",
+                lut.bits_stored, bits_stored
+            );
+        }
+
+        let shift = (bits_stored.next_power_of_two() - (lut.bits_stored as u16)) as u32;
```

Comment:
the if should not check `bits_stored` but `bits_stored.next_power_of_two()` here: it's allowed to have an image with eg. BitsStored=12 but a LUT descriptor that has BitsStored=16 .